ahds/ahds_common.py

```diff
@@ -0,0 +1,791 @@
+# -*- coding: utf-8 -*-
```

Comment:
While I understand the point of this module I'm wondering we can accomplish the same without too much plumbing.

Reply:
I do agree, that is why i wanted you to review first, so that we can identify together where things can be improved and with less achieved the same.